src/codegen/codegen_acc_visitor.cpp

```diff
@@ -57,7 +57,7 @@ void CodegenAccVisitor::print_channel_iteration_block_parallel_hint(BlockType ty
 
 
 void CodegenAccVisitor::print_atomic_reduction_pragma() {
-    if (!info.artificial_cell) {
+    if (info.point_process) {
```

Comment:
Just a `pure nitpick` Ioannis: should we remove this `if` condition from here and put into where this function is called? i.e. `print_atomic_reduction_pragma` should just print the pragma and shouldn't have the extra logic? This way code/usage would be a bit clear?